### ISSUE TITLE
Update TNRIS Lidar dataset & database against new projects in TNRIS S3 bucket

### ISSUE THREAD

**Issue by dhardestylewis** [dhardestylewis/terrain_aggregator#39]
Update TNRIS Lidar dataset & database against new projects in TNRIS S3 bucket

for example, `stratmap21-50cm` is reported in the 2022/05/09 TNRIS availability file but not in `terrain_aggregator` DB

**Comment by dhardestylewis** [comment]
before Aug 10 meeting. include in 2nd grand retiling if possible

**Comment by dhardestylewis** [comment]
[Bexar & Travis Counties Lidar 2021](https://data.tnris.org/collection/447db89a-58ee-4a1b-a61f-b918af2fb0bb)

**Comment by dhardestylewis** [comment]
*updating TACC's TNRIS Lidar DEM collection against the current TNRIS AWS S3 now*

**All TNRIS Lidar DEMs & LPCs can be found at the following location at TACC:**
`stampede2.tacc.utexas.edu:/scratch/projects/tnris/tnris-lidardata`

**Please contact me if you have any issues with accessing the data.**

**The following Lidar collections are being pulled now:**
```
stratmap-2021-28cm-50cm-bexar-travis
stratmap-2021-laguna-madre-bathy
stratmap-2022-50cm-ellis-hill-johnson-navarro
stratmap-2022-address-points
stratmap-2022-land-parcels
usgs-hydrography
utbeg-2017-70cm-laguna-madre
```

Here is how they are being pulled:
```bash
cd /scratch/projects/tnris
while IFS= read -r line ; do aws s3 cp s3://tnris-public-data/production-data/${line}/ tnris-lidardata/${line} --recursive & done < tnris-missing-data.csv
```

`tnris-missing-data.csv` was produced using
```bash
cd /scratch/projects/tnris
aws s3 ls s3://tnris-public-data/production-data/ > tnris-public-data.csv
## use vim magic to modify above file slightly to strip out everything but the collection names
ls tnris-lidardata > tnris-tacc-data.csv
diff --suppress-common-lines -y tnris-public-data.csv tnris-tacc-data.csv
```

<img width="1440" alt="Screen Shot 2022-08-18 at 6 36 18 PM" src="https://user-images.githubusercontent.com/15820710/185513261-081aec36-d456-4a4d-ba01-a03210fd2c60.png">

I have applied the following permissions against this dataset:
```bash
find . -type d -exec bash -c "chgrp G-822180 {} ; chmod 755 {} ; chmod g+s {} ; setfacl -d -m g::rx,o::rx {}" \;
find . -type f -exec bash -c "chgrp G-822180 {} ; chmod 644 {}" \;
```

**Comment by dhardestylewis** [comment]
once downloaded, next step is to upload this data using the instructions in
https://github.com/dhardestylewis/terrain_aggregator#updating-the-database-with-new-lidar-tiles

**Comment by dhardestylewis** [comment]
https://github.com/dhardestylewis/terrain_aggregator/issues/79

- [ ] ⏸️ inclusion of latest TNRIS Lidar DEM tiles paused to return TNRIS, TDIS, & CWE requests for small-scale 1m accuracy results

**Comment by dhardestylewis** [comment]
> _updating TACC's TNRIS Lidar DEM collection against the current TNRIS AWS S3 now_
> 
> **All TNRIS Lidar DEMs & LPCs can be found at the following location at TACC:** `stampede2.tacc.utexas.edu:/scratch/projects/tnris/tnris-lidardata`
> 
> **Please contact me if you have any issues with accessing the data.**
> 
> **The following Lidar collections are being pulled now:**
> 
> ```
> stratmap-2021-28cm-50cm-bexar-travis
> stratmap-2021-laguna-madre-bathy
> stratmap-2022-50cm-ellis-hill-johnson-navarro
> stratmap-2022-address-points
> stratmap-2022-land-parcels
> usgs-hydrography
> utbeg-2017-70cm-laguna-madre
> ```
> 
> Here is how they are being pulled:
> 
> ```shell
> cd /scratch/projects/tnris
> while IFS= read -r line ; do aws s3 cp s3://tnris-public-data/production-data/${line}/ tnris-lidardata/${line} --recursive & done < tnris-missing-data.csv
> ```
> 
> `tnris-missing-data.csv` was produced using
> 
> ```shell
> cd /scratch/projects/tnris
> aws s3 ls s3://tnris-public-data/production-data/ > tnris-public-data.csv
> ## use vim magic to modify above file slightly to strip out everything but the collection names
> ls tnris-lidardata > tnris-tacc-data.csv
> diff --suppress-common-lines -y tnris-public-data.csv tnris-tacc-data.csv > tnris-missing-data.csv
> ## use vim magic to capture the missing collections on the left side of the diff
> ```
> 
> <img alt="Screen Shot 2022-08-18 at 6 36 18 PM" width="1440" src="https://user-images.githubusercontent.com/15820710/185513261-081aec36-d456-4a4d-ba01-a03210fd2c60.png">
> 
> I have applied the following permissions against this dataset:
> 
> ```shell
> cd /scratch/projects/tnris
> find . -type d -exec bash -c "chgrp G-822180 {} ; chmod 755 {} ; chmod g+s {} ; setfacl -d -m g::rx,o::rx {}" \;
> find . -type f -exec bash -c "chgrp G-822180 {} ; chmod 644 {}" \;
> ```

TNRIS Lidar DEM tile collections have been updated successfully!

```bash
cd /scratch/projects/tnris
while IFS= read -r directory ; do echo "${directory} $(~/local/bin/aws/aws s3 ls s3://tnris-public-data/production-data/${directory}/ --recursive | wc -l)" ; done < tnris-missing-data.csv
```
```
stratmap-2021-28cm-50cm-bexar-travis 48922
stratmap-2021-laguna-madre-bathy 189742
stratmap-2022-50cm-ellis-hill-johnson-navarro 13156
stratmap-2022-address-points 19459
stratmap-2022-land-parcels 18532
usgs-hydrography 956
utbeg-2017-70cm-laguna-madre 680
```
```bash
cd /scratch/projects/tnris
while IFS= read -r directory ; do echo "${directory} $(find tnris-lidardata/${directory} -type f | wc -l)" ; done < tnris-missing-data.csv
```
```
stratmap-2021-28cm-50cm-bexar-travis 48922
stratmap-2021-laguna-madre-bathy 189742
stratmap-2022-50cm-ellis-hill-johnson-navarro 13156
stratmap-2022-address-points 19459
stratmap-2022-land-parcels 18532
usgs-hydrography 956
utbeg-2017-70cm-laguna-madre 680
```

<img width="1440" alt="Screen Shot 2022-08-21 at 6 27 39 PM" src="https://user-images.githubusercontent.com/15820710/185815424-95d083ef-1841-44f2-a9f1-fe0ba46c22cf.png">

**Comment by dhardestylewis** [comment]
- [ ] cron job of https://github.com/dhardestylewis/terrain_aggregator/issues/39#issuecomment-1221643453
